pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -209,18 +211,31 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return err
 	}
 	if host == nil {
+		// When finalizer for BHM is removed but an error occur before
+		// the Machine is deleted below in the StateDeleting block, the
+		// Machine should be deleted when no host is found.
+		log.Print("Deleting machine associated with no host: ", machine.Name)
```

Comment:
nit: I'd change this (and the one a few lines below) to "Deleting machine whose associated host is gone." to keep it clear that we don't delete Machines just because they don't have a host.